### PR TITLE
Disable ITF when veteran has a reported date of death

### PR DESCRIPTION
**Pull request by nanotone**
### Description
For ITF-eligible EPs, switch from 040SCRGTY back to 040GTY when veteran is deceased.

Note: previously we suspected that a payee code other than 00 would cause the 040SCRGTY to fail establishment. However, there are a couple instances in prod (EPEs 1190258 and 1202891) with non-00 payee code that _did_ establish, because the veteran is alive.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Added new rspec

### CODE REVIEW

File: app/workflows/end_product_code_selector.rb [hunk 1]
```diff
@@ -305,7 +305,11 @@ def choose_code(end_product_codes)
       END_PRODUCT_CODES[:fiduciary][review_type]
     else
       code = end_product_codes[benefit_type.to_sym][review_type][issue_type]
-      (code == compensation_sc_rating_ep_code && request_issues_older_than_a_year?) ? "040SCRGTY" : code
+      if code == compensation_sc_rating_ep_code && request_issues_older_than_a_year? && veteran.alive?
+        "040SCRGTY"
+      else
+        code
+      end
```
Comment (allthesignals) [inline app/workflows/end_product_code_selector.rb:312]: Seems reasonable to me! :)

**Review by allthesignals — APPROVED** [review]
LGTM!

Question: is EndProductCodeSelector being used as a sort of mixin for a AR Model class? I see it being passed `self` in the the request issue.

**Review by meckila — APPROVED** [review]

**Comment by nanotone** [comment]
@allthesignals I don't think it's really being used as a mixin. The EPCS class is more of a stand-alone utility (for housing all logic related to selecting EP codes) and it just happens to take the RequestIssue as an argument. So when the RequestIssue uses it, it passes `self` -- but this is more of a coïncidence.

**Comment by allthesignals** [comment]
> @allthesignals I don't think it's really being used as a mixin. The EPCS class is more of a stand-alone utility (for housing all logic related to selecting EP codes) and it just happens to take the RequestIssue as an argument. So when the RequestIssue uses it, it passes `self` -- but this is more of a coïncidence.

the passage of `self` makes me think that but yeah that makes sense it's jsut coincidental

**Pull request merged**